presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -539,14 +539,16 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
     private static Schema toIcebergSchema(List<ColumnMetadata> columns)
     {
         List<NestedField> icebergColumns = new ArrayList<>();
+        int index = 1;
```

Comment:
Have you considered making `TypeConverter` an object that stores the index? That should be simpler than passing it through static methods with recursive invocations. Something like
```java
TypeConverter converter = new TypeConverter();
for (ColumnMetadata column : columns) {
    Type type = converter.toIcebergType(column.getType());
    int index = converter.getIndex();
    ...
}
```
We'd probably want to split up the class into `PrestoTypeConverter` and `IcebergTypeConverter` as a precursor commit, since `toPrestoType` can stay static.

Reply:
I made these changes as you suggested. Turns out there is another commit that is not yet pushed upstream that requires this fix. So, I am going to abandon this particular commit and upload a new one when the other commit is pushed upstream (soon).